From 33b613f64e7eff20b6b19fa9edbe56645cb86362 Mon Sep 17 00:00:00 2001
From: William Jordan <bjordan@habana.ai>
Date: Fri, 16 May 2025 21:49:55 +0300
Subject: [PATCH] [SW-218976] add entity aux names to preboot

Add Entity AUX Name PDR to preboot PLDR PDR
database.

Change-Id: I61526e64f00ac6f4e3a700ec5535a24e14abc9d1
Signed-off-by: William Jordan <bjordan@habana.ai>
---

diff --git a/arch/gaudi3/common/ras/pldm/pldm_pdr.h b/arch/gaudi3/common/ras/pldm/pldm_pdr.h
index 72f850d..e4855ca 100644
--- a/arch/gaudi3/common/ras/pldm/pldm_pdr.h
+++ b/arch/gaudi3/common/ras/pldm/pldm_pdr.h
@@ -74,6 +74,9 @@
 	PLDM_RH_SENSOR_BANKS_WITH_MAX_REPLACED_ROWS			= 3027,
 	PLDM_RH_SENSOR_SRAM_BIST_STATUS					= 3055,
 	PLDM_RH_SENSOR_HBM_MBIST_LAST_RUN_STATUS			= 3056,
+	PLDM_RH_ENTITY_AUX_NAMES_ADDIN_CARD				= 4000,
+	PLDM_RH_ENTITY_AUX_NAMES_OAM_CPLD				= 4001,
+	PLDM_RH_ENTITY_AUX_NAMES_ACCELERATOR				= 4002,
 	PLDM_RH_SENSOR_AUX_NAMES_ASIC_SERIAL_NUM_HIGH			= 4011,
 	PLDM_RH_SENSOR_AUX_NAMES_ASIC_SERIAL_NUM_LOW			= 4012,
 	PLDM_RH_SENSOR_AUX_NAMES_DEVICE_PDR_REPOSITORY_VERSION		= 4020,
diff --git a/arch/gaudi3/common/ras/pldm/pldm_pdr_generated.c b/arch/gaudi3/common/ras/pldm/pldm_pdr_generated.c
index a3b86f1..4605ddb 100644
--- a/arch/gaudi3/common/ras/pldm/pldm_pdr_generated.c
+++ b/arch/gaudi3/common/ras/pldm/pldm_pdr_generated.c
@@ -5,12 +5,12 @@
 
 /* !!! AUTO-GENERATED FILE - DO NOT EDIT !!! */
 /*
- * Spreadsheet : '20250331-GAUDI3-RAS-PLDM_7th_stage.xlsx'
- * MD5         : 668acc1ed5a7f3ff192bf5f49bb46efd
+ * Spreadsheet : '20250504_GAUDI3_RAS_PLDM_7th_stage.xlsx'
+ * MD5         : f4a4d7a684a412cfb628b033771747ea
  *
  * Command line:
  *   ./g3-gen-pdrs.py \
- *     -w ".../20250331-GAUDI3-RAS-PLDM_7th_stage.xlsx" \
+ *     -w ".../20250504_GAUDI3_RAS_PLDM_7th_stage.xlsx" \
  *     -r ".../pldm_pdr.h" "pdrs" \
  *     --header ".../pldm_pdr_generated.h" \
  *     --source ".../pldm_pdr_generated.c"
@@ -2641,6 +2641,118 @@
 };
 const size_t pldm_pdr_name_records_length = ARRAY_SIZE(pldm_pdr_name_records);
 
+/**** Auxilary Name PDRs ****/
+
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+
+/* Gaudi3 OAM add-in card accelerator */
+#define _oamAddInCardAccelerator_str "oamAddInCardAccelerator"
+/* Gaudi3 OAM CPLD */
+#define _oamCPLD_str "oamCPLD"
+/* Gaudi3 accelerator */
+#define _accelerator_str "accelerator"
+
+#else /* old aux name format */
+
+/* Gaudi3 OAM add-in card accelerator */
+const uint16_t _oamAddInCardAccelerator_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6f00, 0x6100, 0x6d00, 0x4100, 0x6400, 0x6400,
+	0x4900, 0x6e00, 0x4300, 0x6100, 0x7200, 0x6400,
+	0x4100, 0x6300, 0x6300, 0x6500, 0x6c00, 0x6500,
+	0x7200, 0x6100, 0x7400, 0x6f00, 0x7200, 0x0000
+};
+/* Gaudi3 OAM CPLD */
+const uint16_t _oamCPLD_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6f00, 0x6100, 0x6d00, 0x4300, 0x5000, 0x4c00,
+	0x4400, 0x0000
+};
+/* Gaudi3 accelerator */
+const uint16_t _accelerator_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6100, 0x6300, 0x6300, 0x6500, 0x6c00, 0x6500,
+	0x7200, 0x6100, 0x7400, 0x6f00, 0x7200, 0x0000
+};
+
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+
+struct pldm_pdr_entity_name_record pldm_pdr_entity_name_records[] OFFLOAD_PDR_ATTR = {
+	[PLDM_RIDX_ENTITY_AUX_NAMES_ADDIN_CARD] = {
+		.entity_aux_names = {
+			.header = {
+				.record_handle = PLDM_RH_ENTITY_AUX_NAMES_ADDIN_CARD,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_ENTITY_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.entity_type = PLDM_ENTITY_TYPE_ADDIN_CARD,
+			.entity_instance = 1,
+			.entity_container_id = PLDM_CONTAINER_ADDIN_CARD,
+			.shared_name_count = 0,
+			.names = {
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+				_oamAddInCardAccelerator_str,
+#else /* old aux name format */
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _oamAddInCardAccelerator_str,
+				},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+			},
+		},
+	},
+	[PLDM_RIDX_ENTITY_AUX_NAMES_OAM_CPLD] = {
+		.entity_aux_names = {
+			.header = {
+				.record_handle = PLDM_RH_ENTITY_AUX_NAMES_OAM_CPLD,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_ENTITY_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.entity_type = PLDM_ENTITY_TYPE_CPLD,
+			.entity_instance = 1,
+			.entity_container_id = PLDM_CONTAINER_ADDIN_CARD,
+			.shared_name_count = 0,
+			.names = {
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+				_oamCPLD_str,
+#else /* old aux name format */
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _oamCPLD_str,
+				},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+			},
+		},
+	},
+	[PLDM_RIDX_ENTITY_AUX_NAMES_ACCELERATOR] = {
+		.entity_aux_names = {
+			.header = {
+				.record_handle = PLDM_RH_ENTITY_AUX_NAMES_ACCELERATOR,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_ENTITY_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.entity_type = PLDM_ENTITY_TYPE_ACCELERATOR,
+			.entity_instance = 1,
+			.entity_container_id = PLDM_CONTAINER_ADDIN_CARD,
+			.shared_name_count = 0,
+			.names = {
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+				_accelerator_str,
+#else /* old aux name format */
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _accelerator_str,
+				},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+			},
+		},
+	},
+};
+const size_t pldm_pdr_entity_name_records_length = ARRAY_SIZE(pldm_pdr_entity_name_records);
+
 /**** OEM State Set PDRs ****/
 
 #if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
diff --git a/arch/gaudi3/common/ras/pldm/pldm_pdr_generated.h b/arch/gaudi3/common/ras/pldm/pldm_pdr_generated.h
index 1cc3c13..c3abdfa 100644
--- a/arch/gaudi3/common/ras/pldm/pldm_pdr_generated.h
+++ b/arch/gaudi3/common/ras/pldm/pldm_pdr_generated.h
@@ -5,12 +5,12 @@
 
 /* !!! AUTO-GENERATED FILE - DO NOT EDIT !!! */
 /*
- * Spreadsheet : '20250331-GAUDI3-RAS-PLDM_7th_stage.xlsx'
- * MD5         : 668acc1ed5a7f3ff192bf5f49bb46efd
+ * Spreadsheet : '20250504_GAUDI3_RAS_PLDM_7th_stage.xlsx'
+ * MD5         : f4a4d7a684a412cfb628b033771747ea
  *
  * Command line:
  *   ./g3-gen-pdrs.py \
- *     -w ".../20250331-GAUDI3-RAS-PLDM_7th_stage.xlsx" \
+ *     -w ".../20250504_GAUDI3_RAS_PLDM_7th_stage.xlsx" \
  *     -r ".../pldm_pdr.h" "pdrs" \
  *     --header ".../pldm_pdr_generated.h" \
  *     --source ".../pldm_pdr_generated.c"
@@ -67,6 +67,12 @@
 	PLDM_RIDX_EFFECTER_AUX_NAMES_ETHERNET_PORT_ENABLE,
 };
 
+enum pldm_entity_aux_ridx {
+	PLDM_RIDX_ENTITY_AUX_NAMES_ADDIN_CARD,
+	PLDM_RIDX_ENTITY_AUX_NAMES_OAM_CPLD,
+	PLDM_RIDX_ENTITY_AUX_NAMES_ACCELERATOR,
+};
+
 enum pldm_oem_state_set_ridx {
 	PLDM_RIDX_SENSOR_OEM_STATE_SET_FW_STATE,
 	PLDM_RIDX_SENSOR_OEM_STATE_SET_PPBOOT_IMAGE_SOURCE,
@@ -91,6 +97,9 @@
 extern struct pldm_pdr_name_record pldm_pdr_name_records[];
 extern const size_t pldm_pdr_name_records_length;
 
+extern struct pldm_pdr_entity_name_record pldm_pdr_entity_name_records[];
+extern const size_t pldm_pdr_entity_name_records_length;
+
 extern struct pldm_pdr_oem_state_set_record pldm_pdr_oem_state_set_records[];
 extern const size_t pldm_pdr_oem_state_set_records_length;
 
diff --git a/lib/ras/pldm/pldm.c b/lib/ras/pldm/pldm.c
index 9fb1eb9..a07c053 100644
--- a/lib/ras/pldm/pldm.c
+++ b/lib/ras/pldm/pldm.c
@@ -124,6 +124,8 @@
 		.num_records = pldm_pdr_records_length,
 		.name_records = pldm_pdr_name_records,
 		.num_name_records = pldm_pdr_name_records_length,
+		.entity_name_records = pldm_pdr_entity_name_records,
+		.num_entity_name_records = pldm_pdr_entity_name_records_length,
 		.oem_state_set_records = pldm_pdr_oem_state_set_records,
 		.num_oem_state_set_records = pldm_pdr_oem_state_set_records_length,
 		.oem_unit_records = pldm_pdr_oem_unit_records,
